### PR TITLE
add support for validity-mode expiration-and-subject

### PR DESCRIPTION
**Pull request by Rico29**
#991

### CODE REVIEW

**Comment by Rico29** [comment]
I'm not a perl expert, so I guess it's posssible to put repetitive stuff in subs...

**Comment by Rico29** [comment]
can someone take a look at this please ?

**Comment by Sims24** [comment]
We are going to check you PR, thanks for your contribution. 

Do you have any public site where we can challenge the plugin against ?

**Comment by Rico29** [comment]
Hi Simon
Sorry, you can challenge it on any https enabled website. I don't actually have a website with expired cert or wrong name.

File: apps/protocols/x509/mode/validity.pm [hunk 1]
```diff
@@ -156,6 +156,44 @@ sub run {
...
+                $name = inet_ntoa($name);
+            }
+            if ($name =~ /$self->{option_results}->{subjectname}/mi) {
+                push @subject_matched, $name;
+            } else {
+                $self->{output}->output_add(long_msg => sprintf("Subject Name '%s' is also present in Certificate", $name), debug => 1);
+            }
+        }
+
+        if (@subject_matched == 0) {
```
Comment (Sims24) [inline apps/protocols/x509/mode/validity.pm]: If I understand this one, you check the subject and if it's not OK then you do not try to check expiration date right ?

**Comment by Sims24** [comment]
@Rico29 sounds like a really good idea to merge some checks. 

Would you think that performing every check and filter output is an option for you ? 

I guess we need a deep refactoring on this, in the meantime except the comment above (not a huge fan of skipping checks because other failed ...) I think we can merge this one and we'll refactor it in the future. Just need to check with @cgagnaire if he agrees with me

**Comment by Rico29** [comment]
I totally agree with that ! I'll try to rewrite something next week

**Comment by Sims24** [comment]
Any news here ? Let us know if we can help. 

Rgds

**Comment by CPbN** [comment]
Really nice addidion @cgagnaire, thank you !
One comment perhaps, what about adding a `--timeout` option ?
Thank you :+1:

**Comment by cgagnaire** [comment]
I've just added it (https://github.com/centreon/centreon-plugins/commit/3ece41a4245936ad8773c5ac68c62f7fcb1e3987).
Note that this option will work only with IO::Socket::SSL >= 1.984.
The mode itself will work only with Net::SSLeay >= 1.42.
I've tested latest version of IO::Socket::SSL and Net::SSLeay on both CentOS 6 and 7 and it works well.
We will push those versions on our repos.

**Comment by cgagnaire** [comment]
Anyway, thanks @Rico29 for you work ;)

**Comment by CPbN** [comment]
Many thanks @cgagnaire & @Rico29 :+1:

**Comment by CPbN** [comment]
> The mode itself will work only with Net::SSLeay >= 1.42.
I've tested latest version of IO::Socket::SSL and Net::SSLeay on both CentOS 6 and 7 and it works well.

Strangely enough I face the following issue with CenOS 7 Perl-IO-Socket-SSL package, version 1.94-7 :
`SSL error: SSL wants a read first`

No issue with last version from CPAN, 2.060 (I've then fatpacked it).

> We will push those versions on our repos.

Would then be nice yes :+1:

**Comment by Rico29** [comment]
hello @cgagnaire 
Any news about 
> We will push those versions on our repos.

Regards,
Cédric

**Comment by Rico29** [comment]
@cgagnaire ?

**Comment by CPbN** [comment]
> I've just added it ([3ece41a](https://github.com/centreon/centreon-plugins/commit/3ece41a4245936ad8773c5ac68c62f7fcb1e3987)).
> Note that this option will work only with IO::Socket::SSL >= 1.984.
> The mode itself will work only with Net::SSLeay >= 1.42.
> I've tested latest version of IO::Socket::SSL and Net::SSLeay on both CentOS 6 and 7 and it works well.
> We will push those versions on our repos.

Hi @cgagnaire,
Some news regarding this ?
Many thanks :+1:

**Comment by cgagnaire** [comment]
Hi @Rico29 & @CPbN,
As @garnier-quentin said here https://github.com/centreon/centreon-plugins/issues/1414, we will not update it for now. There is possibility to break other modules.

**Pull request closed without merging**